### PR TITLE
feat: Add 'svn.sourceControl.ignore' option

### PR DESCRIPTION
**Pull request by basecell**
Link #1113

Add `svn.sourceControl.ignore` option, It can ignore unversioned files like .gitignore. At the same time, configuring this option will overlook the default svn ignore rule, but if you do not configure this option, nothing will change.

In addition, I found that when revert multiple added files, if the files are of parent-child structure, the plugin will report an error. To solve this problem just need to reverse the order of the file list.

### CODE REVIEW

**Comment by basecell** [comment]
I found that this pull request has no intention to be merged. Is there any problem?

**Comment by JohnstonCode** [comment]
No problem, just lack of time currently. Will try to have a look this week.

File: src/commands/revert.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export class Revert extends Command {
         return;
       }
 
-      const paths = resources.map(resource => resource.fsPath);
+      const paths = resources.map(resource => resource.fsPath).reverse();
```
Comment (JohnstonCode) [inline src/commands/revert.ts:29]: Nice catch!

File: src/commands/revertAll.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export class RevertAll extends Command {
         return;
       }
 
-      const paths = resources.map(resource => resource.fsPath);
+      const paths = resources.map(resource => resource.fsPath).reverse();
```
Comment (JohnstonCode) [inline src/commands/revertAll.ts:29]: Nice catch!

**Review by JohnstonCode — APPROVED** [review]
Looks good!

**Comment by basecell** [comment]
Thanks.

**Pull request merged**